private_gpt/components/ingest/ingest_helper.py

```
@@ -33,6 +37,60 @@ def transform_file_into_documents(
+        Args:
+            pdf_path (Path): Path to the pdf file
+            add_summary (bool, optional): Add a summary of the entire pdf. Defaults to True.
+
+        Returns:
+            list[schema.Document]: List of documents
+        """
+        logger.info("Custom - Transforming file_name=%s into documents", file_name)
+
+        try:
+            filename = file_data
+            parsed_text = partition_pdf(filename, multipage_sections=False)
+            # First ensure all chunks relate to a single page
+            text_by_page = {}
+            for doc in parsed_text:
+                page_number = doc.metadata.page_number
+                if page_number not in text_by_page:
+                    text_by_page[page_number] = ""
+                text_by_page[page_number] += f"{doc.text} "
+
+            # Now convert to documents and manually set some of the metadata
+            documents = []
+            for page_number in text_by_page:
+                text_content = text_by_page[page_number]
+                llamaindex_doc = Document(
+                    text=text_content,
+                    metadata={
+                        "filename": filename.name,
+                        "page_number": page_number,
+                    },
+                )
+                documents.append(llamaindex_doc)
+
+            if add_summary:
+                summary_document = IngestionHelper._summarize_entire_pdf(
+                    filename=filename, verbose=False
+                )
+                documents = [summary_document] + documents
+
+            for document in documents:
+                document.metadata["file_name"] = file_name
+            # IngestionHelper._exclude_metadata(documents) - TODO: determine if we need this step
+            return documents
+
+        except Exception as error:
+            logger.error(f"ingest_helper - Error loading pdf document: {error}")
+            raise error
```

Keeping the entire service logic in a catch-all exception block makes it difficult to debug errors. I highly recommend removing this try/catch block and letting the exceptions raise where they would happen.

If any custom error handling logic should happen, they should be attached to a single operation.